internal/store/postgres/billing_transactions_repository.go

```diff
@@ -388,9 +422,28 @@ func (r BillingTransactionRepository) getBalanceInTx(ctx context.Context, tx *sq
 // in transaction table till now.
 func (r BillingTransactionRepository) GetBalance(ctx context.Context, accountID string) (int64, error) {
 	var amount int64
-	if err := r.dbc.WithTxn(ctx, sql.TxOptions{}, func(tx *sqlx.Tx) error {
+	if err := r.dbc.WithTxn(ctx, sql.TxOptions{
+		Isolation: sql.LevelSerializable,
+	}, func(tx *sqlx.Tx) error {
+		var err error
+		amount, err = r.getBalanceInTx(ctx, tx, accountID, nil, nil)
+		return err
+	}); err != nil {
+		return 0, fmt.Errorf("failed to get balance: %w", err)
+	}
+	return amount, nil
+}
+
+// GetBalanceForRange returns the balance of the account in the given range.
```

Comment:
Per our discussion, we had decided to keep a balance for the start of the month, and calculate `end balance - start balance` to figure out if there is an overdraft (similar to a banking system).
If we don't keep the starting balance, won't we end up with incorrect overdrafts? For example, if I already had 1000 tokens in my account at the start of September (from token additions in previous months). And in September I spend 200 tokens. It would appear that my balance for September is `-200`, but it is actually `+800` due to earlier balance. 

Plainly going off the comment. Let me know in case I've understood it incorrectly.

Reply:
Discussed offline